extensions/amp-list/0.1/amp-list.js

```diff
@@ -1109,6 +1170,48 @@ export class AmpList extends AMP.BaseElement {
     }
   }
 
+  /**
+   * Measure and lock height before performing given mutate fn.
+   * Applicable for amp-list initialized with layout=container.
+   * @private
+   * @param {function():(Promise|undefined)} mutate
+   * @return {!Promise}
+   */
+  lockHeightAndMutate_(mutate) {
+    if (!this.enableManagedResizing_ || this.loadMoreEnabled_) {
+      dev().error(
+        TAG,
+        '%s initialized with layout=container does not support infinite scrolling with [load-more]. %s',
+        this.element
+      );
+      return Promise.resolve();
+    }
+    let currentHeight;
+    return this.measureMutateElement(
+      () => {
+        currentHeight = this.element./*OK*/ offsetHeight;
+      },
+      () => {
+        setImportantStyles(this.element, {
+          'height': `${currentHeight}px`,
+          'overflow': 'hidden',
+        });
+        return mutate();
+      }
+    );
+  }
+
+  /**
+   * Applicable for amp-list initialized with layout=container.
+   * @private
+   */
+  unlockHeightInsideMutate_() {
+    setImportantStyles(this.element, {
```

Comment:
Optional: You can add a `devAssert` here to make sure future devs don't use call this inappropriately.